client/src/components/SpendingInsights.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { currencyFilter } from '../util';
+import { CategoriesChart } from '.';
+
+SpendingInsights.propTypes = {
+  transactions: PropTypes.array,
+};
+
+export default function SpendingInsights({ transactions }) {
+  // grab transactions from most recent month and filter out transfers and payments
+  const today = new Date();
+  const oneMonthAgo = new Date().setDate(today.getDate() - 30);
+  const monthlyTransactions = transactions.filter(tx => {
+    const date = new Date(tx.date);
+    return (
+      date > oneMonthAgo &&
+      tx.category !== 'Payment' &&
+      tx.category !== 'Transfer'
+    );
+  });
+
+  // create category and name objects from transactions
+  const categories = {};
+  monthlyTransactions.forEach(tx => {
+    categories[tx.category] = (categories[tx.category] || 0) + tx.amount;
+  });
+
+  const names = {};
+  monthlyTransactions.forEach(tx => {
+    names[tx.name] = (names[tx.name] || 0) + tx.amount;
+  });
+
+  // sort names by spending totals
+  var sortableNames = [];
+  for (const name in names) {
+    sortableNames.push([name, names[name]]);
+  }
+  sortableNames.sort((a, b) => b[1] - a[1]);
+  sortableNames.splice(5); // top 5
```

Comment:
I dont think it makes a big difference, but its just good practice to put complicated calculations like the ones here between line 15 and 41 into a `useMemo`. If you use a useMemo with the correct dependencies, these values will only be re-calculated when the data, such as `transactions` changes instead of being recalculated on every render

Reply:
Also as discussed in DM, a lot of this logic could move to the backend